### ISSUE TITLE
[Bug]: Interactive mode - plot not displaying in interactive mode after calling `input()`

### ISSUE THREAD

**Issue by lostinnowhere314** [matplotlib/matplotlib#25930]
### Bug summary

In interactive mode (`plt.ion()`), creating a plot and then immediately having a call to `input()` causes the plot to not display until after the call to `input()` has finished. Specifically, only an empty window is created initially, and it is not populated until after the call to `input()` is finished.

### Code for reproduction

```python
from matplotlib import pyplot as plt
x = [0, 1]
y = [2, 3]
plt.ion()
# We expect the call to plt.plot() to create a figure since we're in interactive mode:
plt.plot(x, y); plt.input("Some input: ")

# This line has the same effect as the above one:
plt.plot(x, y); plt.ioff(); plt.show(block=False); input("Some input: ")
```

### Actual outcome

Matplotlib produces the following empty window:
![image](https://github.com/matplotlib/matplotlib/assets/78163515/452fb4f8-c69c-4986-9140-4e05c97fb3b4)
Attempting to interact with this window tends to make Python unhappy.

The main terminal is pefectly responsive, and the input prompt can be typed in.
Only after the `input()` call is complete will the plot actually display as intended, even with the call.

### Expected outcome

Ideally, the plot would update and display while `input()` is waiting for user input.

### Additional information

Tested on several versions back; this issue also occurs on version 3.7.0, but not on versions 3.6.0 to 3.6.3.

Some additional info:
- This issue happens in scripts and in the interactive terminal (ran in IPython 7.34.0), but not in Jupyter notebooks (tested the same snippet and it works the expected way).
- If a plot is already displaying in interactive mode and this happens, the plot will just not update until after `input()` is finished
- This also happens if the creation of the plot and the call to `input` are more widely-separated; calling `time.sleep()` in between does not help.
- Turning off interactive mode before calling `input` does not fix the issue, nor does calling `plt.show()` with `block=False` (f.e. the line `plt.plot(x, y); plt.ioff(); plt.show(block=False); input("Some input: ")` runs exactly the same)

### Operating system

Windows 10

### Matplotlib Version

3.7.1

### Matplotlib Backend

TkAgg

### Python version

3.9.12

### Jupyter version

6.4.8

### Installation

pip

**Comment by tacaswell** [comment]
To understand what is going on here, please see https://matplotlib.org/stable/users/explain/interactive_guide.html .

TL/DR: for a UI to be responsive there needs to be an (infinite) loop running to run the code to make it live.  The InputHook will run the GUI event loop while waiting from the slow human to type which makes it look like both the prompt and the GUI are responsive, but what is really happening is that it is switching faster than you can see.

What is going on here is that when you are in `input` the input hook is not running (as to exactly why that is, it will depend on which shell you are using and which backend).  The notebook one works because the blocking nature of the input is shifted to the browser so you escape this problem (and the kernel runs the input hook while it waits for a message back over the socket).

Looking at https://github.com/python/cpython/pull/7978#issuecomment-401059355 exactly this behavior is described, however no one ever followed up to make a new issue so I think this is the expected behavior on windows for the plain Python prompt.  I suspect if you `Ipython` and use prompt-toolkit it will work (but I have not tested that).

**Comment by timhoffm** [comment]
Not reproducible here either.
